sdk/core/loader/boot.cc

```diff
@@ -435,6 +502,51 @@ namespace
 			return buildMMIO();
 		}
 
+		// Privileged compartments don't have sealed objects.
+		if constexpr (!std::is_same_v<
+		                std::remove_cvref_t<decltype(sourceCompartment)>,
+		                ImgHdr::PrivilegedCompartment>)
+		{
+			if (contains(sourceCompartment.sealedObjects, target, size))
+			{
+				auto sealingType =
+				  build<uint32_t,
+				        Root::Type::RWGlobal,
+				        PermissionSet{Permission::Load, Permission::Store}>(
+				    target);
+				// TODO: This currently places a restriction that data memory
+				// can't be in the low 64 KiB of the address space.  That may be
+				// too restrictive. If we haven't visited this sealed object
```

Comment:
If we haven't visited yet?  That seems a little worrying as an approach.  From our discussion, I thought we could get away with two passes without the need to re-visit anything?

Reply:
Two import entries may point to the same cap (they probably shouldn't but I didn't want to preclude it).